### PR TITLE
Cleanup: move to correct phpunit assertions

### PR DESCRIPTION
**Pull request by Ocramius**
As discussed in #110

### CODE REVIEW

**Comment by lcobucci** [comment]
Should we also change `$this->assert*` to `self::assert*` here or on a different PR?

**Comment by Ocramius** [comment]
@lcobucci good point - will apply that change too :-)

File: tests/Doctrine/Tests/Common/Annotations/Fixtures/TestInterface.php [hunk 1]
```diff
@@ -9,5 +9,5 @@
     /**
      * @Secure
      */
-    function foo();
+    public function foo();
 }
```
Comment (lcobucci) [inline tests/Doctrine/Tests/Common/Annotations/Fixtures/TestInterface.php]: Should we have the newline at the end?

File: tests/Doctrine/Tests/Common/Annotations/Fixtures/InterfaceWithConstants.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace Doctrine\Tests\Common\Annotations\Fixtures;
+
+interface InterfaceWithConstants
+{
+
+    const SOME_VALUE = 'InterfaceWithConstants.SOME_VALUE';
+    const SOME_KEY   = 'InterfaceWithConstants.SOME_KEY';
+}
```
Comment (lcobucci) [inline tests/Doctrine/Tests/Common/Annotations/Fixtures/InterfaceWithConstants.php]: Here as well

File: tests/Doctrine/Tests/Common/Annotations/DocParserTest.php [hunk 3]
```diff
@@ -653,15 +653,15 @@ public function testAnnotationWithRequiredAttributes()
 
         $docblock   = '@Doctrine\Tests\Common\Annotations\Fixtures\AnnotationWithRequiredAttributes("Some Value")';
         try {
-            $result = $parser->parse($docblock,$context);
+            $parser->parse($docblock,$context);
```
Comment (lcobucci) [inline tests/Doctrine/Tests/Common/Annotations/DocParserTest.php]: Shouldn't we have a space after the comma?

File: tests/Doctrine/Tests/Common/Annotations/DocParserTest.php [hunk 4]
```diff
@@ -653,15 +653,15 @@ public function testAnnotationWithRequiredAttributes()
...
             $this->fail();
         } catch (AnnotationException $exc) {
             $this->assertContains('Attribute "annot" of @Doctrine\Tests\Common\Annotations\Fixtures\AnnotationWithRequiredAttributes declared on property SomeClassName::invalidProperty. expects a(n) Doctrine\Tests\Common\Annotations\Fixtures\AnnotationTargetAnnotation. This value should not be null.', $exc->getMessage());
         }
 
         $docblock   = '@Doctrine\Tests\Common\Annotations\Fixtures\AnnotationWithRequiredAttributes(annot = @Doctrine\Tests\Common\Annotations\Fixtures\AnnotationTargetAnnotation)';
         try {
-            $result = $parser->parse($docblock,$context);
+            $parser->parse($docblock,$context);
```
Comment (lcobucci) [inline tests/Doctrine/Tests/Common/Annotations/DocParserTest.php]: Shouldn't we have a space after the comma?

File: tests/Doctrine/Tests/Common/Annotations/DocParserTest.php [hunk 5]
```diff
@@ -687,15 +687,15 @@ public function testAnnotationWithRequiredAttributesWithoutConstructor()
 
         $docblock   = '@Doctrine\Tests\Common\Annotations\Fixtures\AnnotationWithRequiredAttributesWithoutConstructor("Some Value")';
         try {
-            $result = $parser->parse($docblock,$context);
+            $parser->parse($docblock,$context);
```
Comment (lcobucci) [inline tests/Doctrine/Tests/Common/Annotations/DocParserTest.php]: Shouldn't we have a space after the comma?

File: tests/Doctrine/Tests/Common/Annotations/DocParserTest.php [hunk 6]
```diff
@@ -687,15 +687,15 @@ public function testAnnotationWithRequiredAttributesWithoutConstructor()
...
             $this->fail();
         } catch (AnnotationException $exc) {
             $this->assertContains('Attribute "annot" of @Doctrine\Tests\Common\Annotations\Fixtures\AnnotationWithRequiredAttributesWithoutConstructor declared on property SomeClassName::invalidProperty. expects a(n) \Doctrine\Tests\Common\Annotations\Fixtures\AnnotationTargetAnnotation. This value should not be null.', $exc->getMessage());
         }
 
         $docblock   = '@Doctrine\Tests\Common\Annotations\Fixtures\AnnotationWithRequiredAttributesWithoutConstructor(annot = @Doctrine\Tests\Common\Annotations\Fixtures\AnnotationTargetAnnotation)';
         try {
-            $result = $parser->parse($docblock,$context);
+            $parser->parse($docblock,$context);
```
Comment (lcobucci) [inline tests/Doctrine/Tests/Common/Annotations/DocParserTest.php]: Shouldn't we have a space after the comma?

File: tests/Doctrine/Tests/Common/Annotations/DocParserTest.php (hunk 4 above)
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Annotations/DocParserTest.php]: I didn't apply any CS fixes

**Comment by Ocramius** [comment]
@lcobucci if you think it's good to go, go on and 🚢

File: tests/Doctrine/Tests/Common/Annotations/AbstractReaderTest.php [hunk 7]
```diff
@@ -20,90 +20,90 @@ public function testAnnotations()
         $class = $this->getReflectionClass();
         $reader = $this->getReader();
 
-        $this->assertCount(1, $reader->getClassAnnotations($class));
-        $this->assertInstanceOf($annotName = DummyAnnotation::class, $annot = $reader->getClassAnnotation($class, $annotName));
-        $this->assertEquals('hello', $annot->dummyValue);
+        self::assertCount(1, $reader->getClassAnnotations($class));
```
Comment (mikeSimonson) [inline tests/Doctrine/Tests/Common/Annotations/AbstractReaderTest.php]: Can you point to the reasoning for this change or explain it ?

File: tests/Doctrine/Tests/Common/Annotations/Fixtures/Controller.php [hunk 8]
```diff
@@ -192,7 +192,7 @@ private function updateFieldAceProperty($name, array $changes)
 
                     $aceIdProperty = new \ReflectionProperty('Symfony\Component\Security\Acl\Domain\Entry', 'id');
                     $aceIdProperty->setAccessible(true);
-                    $aceIdProperty->setValue($ace, intval($aceId));
+                    $aceIdProperty->setValue($ace, (int) $aceId);
```
Comment (mikeSimonson) [inline tests/Doctrine/Tests/Common/Annotations/Fixtures/Controller.php]: What is the reason for the preference ?

File: tests/Doctrine/Tests/Common/Annotations/AbstractReaderTest.php (hunk 7 above)
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Annotations/AbstractReaderTest.php]: Isn't it equivalent?

File: tests/Doctrine/Tests/Common/Annotations/Fixtures/Controller.php (hunk 8 above)
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Annotations/Fixtures/Controller.php]: Intval is generally to be replaced with simpler to look up casts

File: tests/Doctrine/Tests/Common/Annotations/AbstractReaderTest.php (hunk 7 above)
Comment (mikeSimonson) [inline tests/Doctrine/Tests/Common/Annotations/AbstractReaderTest.php]: Yes it is (as far as I know) that why I want to know why you @lcobucci preferer this version.

**Comment by mikeSimonson** [comment]
Also how am I suppose to validate mark the review as being positive (without requiring change)? I can't find any button :cry: .

**Comment by Ocramius** [comment]
Only while the PR is still open

**Comment by mikeSimonson** [comment]
Thanks :+1: anyway

File: tests/Doctrine/Tests/Common/Annotations/AbstractReaderTest.php (hunk 7 above)
Comment (mikeSimonson) [inline tests/Doctrine/Tests/Common/Annotations/AbstractReaderTest.php]: Wait I was talking about the change from $this->assertCount to $self::assertCount not the change from assertTrue( , count()) to assertCount(). Or it's the same answer ?

**Comment by Ocramius** [comment]
Oh, that's just because static methods should be invoked statically,
nothing else

**Pull request merged**